contracts/MultiSigWallet.sol

```diff
@@ -166,334 +260,399 @@ contract MultiSigWallet {
+                break;
+            }
         }
-        confirmations[_transactionId].status = 1;
-
-        emit Proposed(
-            _transactionId,
-            msg.sender
-        );
+        isWallet[_oldWallet] = false;
+        isWallet[_newWallet] = true;
     }
 
     /**
-     * @notice It is used to send the transaction to confirmation state by the
-     *         wallet who has sent the transaction.
+     * @notice Changes requirement.
      *
-     * @param _transactionId It marks address(this) transaction id as confirmed against
-     *        the wallet which has sent the transaction.
+     * @dev Function requires:
+     *          - Transaction is sent by multisig.
+     *          - Requirement validity held.
+     *
+     * @param _required The number of required confirmations.
+     *
+     * @return transactionId_ of the proposal.
      */
-    function confirmTransaction(
-        bytes32 _transactionId
+    function changeRequirement(
+        uint256 _required
     )
-        internal
+        public
+        onlyMultisig
+        validRequirement(wallets.length, _required)
     {
-        require(
-            confirmations[_transactionId].status == 1,
-            "Please first propose the transaction"
-        );
-        require(
-            !confirmations[_transactionId].isConfirmedBy[msg.sender],
-            "Transaction is already confirmed by this wallet"
-        );
+        required = _required;
+    }
 
-        confirmations[_transactionId].isConfirmedBy[msg.sender] = true;
-        emit Confirmed(
-            _transactionId,
-            msg.sender
+    /** @notice Returns the number of registered wallets. */
+    function walletCount()
+        public
+        view
+        returns(uint256)
+    {
+        return wallets.length;
+    }
 
-        );
 
-        if (isAlreadyProposedTransaction(_transactionId)) {
-            if (isRequirementAchieved(_transactionId)) {
-                confirmations[_transactionId].status = 2;
-            }
-        }
-    }
+    /* Internal Functions */
 
     /**
-     * @notice  It is used to check whether the transaction is in proposed
-     *          state.
+     * @notice Returns the confirmation status of a transaction.
      *
-     * @param _transactionId It denotes whose transaction status
-                (Proposed state) is to be checked.
+     * @dev Transaction is confirmed if wallets' count that confirmed
+     *      the transaction is bigger or equal to required.
+     *      Function requires:
+     *          - Transaction with the specified id exists.
      *
-     * @return true if transaction is in proposed state.
+     * @param _transactionID Transaction id to check.
+     *
+     * @return Returns true in case if transaction confirmed, otherwise
+     *         false.
      */
-    function isAlreadyProposedTransaction(
-        bytes32 _transactionId
-    )
+    function isTransactionConfirmed(uint256 _transactionID)
         internal
```

Comment:
I think we should add note the reason for making this function "Internal" so that in future if anyone tries to make it public/external they should be aware of the replaceWallet and removeWallet scenarios.